### PR TITLE
Add support for automatically monitoring consul-configured values

### PR DESCRIPTION
**Pull request by aziule**
## Which problem is this PR solving?

Solves #58 

## Short description of the changes

Add the ability to automatically monitor Consul changes from the config, without having to pass any additional item.

### CODE REVIEW

File: harvester.go [hunk 1]
```diff
@@ -5,6 +5,7 @@ import (
 	"time"
 
 	"github.com/beatlabs/harvester/config"
+	harvesterlog "github.com/beatlabs/harvester/log"
```
Comment (mantzas) [inline harvester.go:8]: no need to alias here, no colission.

File: harvester.go [hunk 2]
```diff
@@ -98,6 +99,25 @@ func (b *Builder) WithConsulMonitor(addr, dc, token string, timeout time.Duratio
 	return b
 }
 
+// WithConsulMonitorFromConfig enables support for monitoring key/prefixes on consul.
+//
+// It differs from WithConsulMonitor as it automatically parses the config and monitors every field tagged with consul.
+func (b *Builder) WithConsulMonitorFromConfig(addr, dc, token string, timeout time.Duration) *Builder {
```
Comment (mantzas) [inline harvester.go:105]: Since we have WithConsul, why we do not reuse this one. This would make the API clear and one way..

Comment (aziule) [reply harvester.go:105]: Do you suggest to replace `WithConsulMonitor` entirely and only keep one method?

I didn't want to introduce breaking changes but I'm fine with it.

Comment (aziule) [reply harvester.go]: Fixed - please double check

File: harvester.go (hunk 1 above)
Comment (aziule) [reply harvester.go]: Fixed

**Pull request merged**